add 10% chance to execute ls

red-socks294 opened the pull request:
because of the StackExchange thread "[Does sl ever show the current directory?](https://unix.stackexchange.com/questions/124816/does-sl-ever-show-the-current-directory)"

red-socks294 committed:
Update sl.c

Julien-Leos commented:
I approve. Now please add 1% chance on ls to execute sl.

red-socks294 commented:
Okay... but i'll first attempt to check if sl is indeed installed on the system

tralph3 commented:
I would suggest running `ls` and letting the shell interpret it, that way aliases should be respected.

red-socks294 commented:
I will do that when I get to my PC.

okalachev commented:
This would ruin the joke:

>As far as I know, the only condition under which sl shows the current directory is when you mistype it as ls.

So I'm against.

Commit:
Update it so it has a 10% chance to ls

Update it so it has a 10% chance to execute ls, and if it does do it it will /bin/bash the ls.

red-socks294 commented:
> I would suggest running `ls` and letting the shell interpret it, that way aliases should be respected.

This has been added.

yuantj commented on sl.c:
```diff
@@ -83,7 +83,12 @@ void option(char *str)
+    if(rand() % 100 < 10)
+    {
+        /*10% chance to execute ls*/
+        execv("/bin/bash -c \"/bin/ls\"", argv);
```
```suggestion
        execvp("ls", argv);
```

Use `execvp` to avoid dependency of `bash`.

red-socks294 replied:
done

Commit:
Update sl.c
